### ISSUE TITLE
instagram: handle new user-provided image alt text

### ISSUE THREAD

**Issue by snarfed** [snarfed/granary#159]
https://instagram-press.com/blog/2018/11/28/creating-a-more-accessible-instagram/

**Comment by snarfed** [comment]
evidently this is surfaced in the `edge_sidecar_to_children.edges.node.accessibility_caption` field. https://chat.indieweb.org/dev/2019-01-10#t1547144886949000

also, problem, if there's no user-provided alt text, instagram puts auto-generated text in that same field, either _No photo description available._ or _Image may contain: ..._ (from ML image processing). [we haven't yet found a way to distinguish user-provided from auto-generated](https://chat.indieweb.org/dev/2019-01-10#t1547146231409900), so we may have to use string matching for that. 😢

**Comment by snarfed** [comment]
example post with user-provided alt text on two photos: https://www.instagram.com/p/Bq8U12UAcdq/

**Comment by fridastovern** [comment]
am I the only one that experience problems with this? Alt text works fine when I'm trying it on my own account, but when my friends tries it from their accounts it does not read the alt text with VoiceOver. Is this a big bug in the system?

**Comment by snarfed** [comment]
@fridastovern sorry for the trouble!

looks like granary is still including alt text with instagram photos if it was directly provided by the user. (ie not auto-generated alt text like "Image may contain: one person, closeup.") example: https://granary.io/instagram/snarfed/@self/@app/BsdlOmLh_IX?format=html

do you have a specific granary URL that doesn't work for your friends? happy to look if so.

**Comment by fridastovern** [comment]
Hm, I did not get that a 100%. I have added alt text, written text like "the street is covered with snow" The profile is open, and if function fine when I'm using my account. It's when I'm switching and trying from my other accounts the problem is occurring.  The strange thing is that your picture worked fine.

You can try, the text will be in Norwegian so it will probably sound weird. maybe that's the problem? 

https://www.instagram.com/designinblindness

Thank you so much :-)

**Comment by snarfed** [comment]
thanks for the example! hmm, looking at https://granary.io/instagram/designinblindness/@self/@app/?format=html , all of the images have alt text.

where are you seeing this problem? in a feed reader that's subscribed to a granary feed? if so, could you post the exact feed URL here, and which feed reader you're using?

(btw, if you're seeing this problem inside instagram, then sadly i can't help you with that, since instagram itself doesn't use granary.)

**Comment by fridastovern** [comment]
aha, that's probably the problem. I'm just using instagram with VoiceOver.
